### PR TITLE
feat(wasm): export an autocomplete function

### LINKED ISSUE #3111: wasm: autocomplete function
Export functions to interact with CLI autocomplete. If possible the AutoCompleteTree should be generated only one time.

### LINKED ISSUE #3114: wasm: correct typings
The types in the wasm package should be improved an cleaned up.

An `index.d.ts` can be created at the package root, it can be added to package.json as `"types": "index.d.ts"`.

This type index should export all types, the ones for wasm_exec, the CLI and the current index.js

### PR DESCRIPTION
**Pull request by Codelax**
Closes #3111 
Closes #3114 
Currently not using directly autocomplete as it requires a filled Context, will be done in the future.

The completion tree is built upon each call.

### CODE REVIEW

File: internal/wasm/autocomplete.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	RightWords   []string `js:"rightWords"`
+}
+
+func Autocomplete(cfg *AutoCompleteConfig) ([]string, error) {
+	indexToComplete := int64(len(cfg.LeftWords) + 2)
+
+	words := append(cfg.LeftWords, cfg.SelectedWord)
+	words = append(words, cfg.RightWords...)
+
+	completeCommand := []string{"autocomplete", "complete", "zsh", strconv.FormatInt(indexToComplete, 10), "scw"}
```
Comment (remyleone) [inline internal/wasm/autocomplete.go:22]: Could ZSH be in the AutoCompleteConfig? Maybe depending on the different feature of a shell we want to configure them differently?

File: wasm/src/jshelpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (remyleone) [inline wasm/src/jshelpers.test.ts:1]: Is there a linter running on all the typescript code that we got?

Comment (Codelax) [reply wasm/src/jshelpers.test.ts:1]: Not currently

File: internal/wasm/autocomplete.go (hunk 1 above)
Comment (remyleone) [reply internal/wasm/autocomplete.go:22]: Could you add a comment about this directly in the code to give directions about how to refactor this code if you think it is going to be complex and time-consuming?

File: wasm/src/jshelpers.test.ts (hunk 2 above)
Comment (remyleone) [reply wasm/src/jshelpers.test.ts:1]: https://github.com/scaleway/scaleway-cli/issues/3154

**Review by remyleone — APPROVED** [review]

**Pull request merged**